packages/toolpad-app/src/appDom/index.ts

```diff
@@ -1114,15 +1114,15 @@ export function ref(nodeId: NodeId): NodeReference;
 export function ref(nodeId: null | undefined): null;
 export function ref(nodeId: Maybe<NodeId>): NodeReference | null;
 export function ref(nodeId: Maybe<NodeId>): NodeReference | null {
-  return nodeId ? { $$ref: nodeId } : null;
+  return nodeId ? { $ref: nodeId } : null;
```

Comment:
When we inject the initial DOM state in the HTML in the `<script>` tags all the `$$ref` properties are being changed to `$ref` now with the Vite runtime... Not sure exactly why.
Can we just use `$ref` for these properties? It shouldn't even need a migration, right? I only saw it being used in a few old connections besides navigation actions.

Reply:
We currently have two formats for bindings/actions/...

One legacy one in the appDom, one new one from the new file system format. Over time we will move the whole application over to the second format

Reply:
Ok! How do you suggest fixing the navigation actions for now? `appDom.deRef` tries to get `$$ref` but the local DOM has the page as `$ref`.
Are the changes in this PR about that ok or do you suggest an alternative?

Reply:
Since you're changing the existing format, my suggestion would be to change it to align it close to the what's in the file schema. The more we can reduce the amount of conversion needed from schema to internal representation, the better. With the north star goal to get rid of the conversion completely.
Otherwise, if it works, it works, and this can be merged

Reply:
I forgot this comment... will take a second look and add changes if it makes sense.